### PR TITLE
[#187364736] Admin can disable account for various reasons

### PR DESCRIPTION
**Pull request by mpanoathos**
#### What does this PR do?
Disabling account for different reasons
#### Description of Task to be completed?
Admin will be able to disable account for various reasons by changing user status
#### How should this be manually tested?
1. Using postman
2. Swagger 
#### What are the relevant pivotal tracker/Trello stories?
#187364736
#### Screenshots (if appropriate)
https://www.loom.com/share/20aefa149c344395a3905df58ebcc6d2?sid=1727ecb4-731e-4f22-bf93-74b6aaa3c2d9

### CODE REVIEW

**Review by Impano1 — APPROVED** [review]

File: src/middlewares/authenticationMiddleware.ts [hunk 1]
```diff
@@ -33,11 +33,11 @@ const isAuthenticated = async (
     next()
   } catch (error) {
     res.status(500).json({ message: 'Internal server down' })
+    console.log(error)
```
Comment (Yvan-David) [inline src/middlewares/authenticationMiddleware.ts]: I think you shouldn't have touched to this file as there is no new changes to be added
and console.log() is not supported by eslint try to replace it

**Review by Yvan-David — CHANGES_REQUESTED** [review]
Please try to provide the req.body that provides reasons for account suspension

Also an email should be send to the user after the account is disabled

**Review by Yvan-David — CHANGES_REQUESTED** [review]
you did a great job but they are some files you changed that didn't need to be changed like loginController and authenticationMiddleware please revert the changes to reduce risks of conflicts

**Review by Yvan-David — APPROVED** [review]
Great job on working on the reviews and also the whole feature

File: test/changeStatus.test.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    // it('should handle email sending errors gracefully', async () => {
+    //   const user = { status: 'active', save: sinon.stub().resolves(), username: 'testuser', email: 'test@example.com' } as any;
+    //   userStub.resolves(user);
+
+    //   await adminController.updateStatus(req as Request, res as Response, 'inactive');
+
+    //    expect(res.status).to.have.been.calledWith(500);
+    // });
+  });
```
Comment (UwizeyimamaFurahaJustine) [inline test/changeStatus.test.ts:80]: If you are not using this test its better to remove it instead of commenting it.

**Review by Furaha-Justine — COMMENTED** [review]
You can work on that request, otherwise you have done a good work.

File: test/changeStatus.test.ts (hunk 2 above)
Comment (Furaha-Justine) [inline test/changeStatus.test.ts]: If you are not using this test its better to remove it instead of commenting it

**Review by Furaha-Justine — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**